test/acceptance/test/utils.go

```diff
@@ -624,3 +605,85 @@ func takeScreenshot() string {
 	}
 	return ""
 }
+
+func getWaitTimeFromErr(errOutput string) (time.Duration, error) {
+	var re = regexp.MustCompile(`(?m)\[rate reset in (.*)\]`)
+	match := re.FindAllStringSubmatch(errOutput, -1)
+
+	if len(match) >= 1 && len(match[1][0]) > 0 {
+		duration, err := time.ParseDuration(match[1][0])
+		if err != nil {
+			return 0, fmt.Errorf("error pasing rate reset time %w", err)
+		}
+		return duration, nil
+	}
+
+	return 0, fmt.Errorf("could not found a rate reset on string: %s", errOutput)
+}
+
+func createRepository(repoName string, private bool) error {
+
+	visibility := gitprovider.RepositoryVisibilityPrivate
+	if !private {
+		visibility = gitprovider.RepositoryVisibilityPublic
+	}
+
+	description := "Weave Gitops test repo"
+	defaultBranch := "main"
+	repoInfo := gitprovider.RepositoryInfo{
+		Description:   &description,
+		Visibility:    &visibility,
+		DefaultBranch: &defaultBranch,
+	}
+
+	repoCreateOpts := &gitprovider.RepositoryCreateOptions{
+		AutoInit: gitprovider.BoolVar(true),
+	}
+
+	orgRef := gitprovider.OrgRepositoryRef{
+		RepositoryName: repoName,
+		OrganizationRef: gitprovider.OrganizationRef{
+			Domain:       github.DefaultDomain,
+			Organization: GITHUB_ORG,
+		},
+	}
+
+	githubProvider, err := github.NewClient(
+		gitprovider.WithOAuth2Token(os.Getenv("GITHUB_TOKEN")),
+		gitprovider.WithDestructiveAPICalls(true),
+	)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	fmt.Printf("creating repo %s ...\n", repoName)
+	if err := utils.WaitUntil(os.Stdout, time.Second, time.Second*30, func() error {
+		_, err := githubProvider.OrgRepositories().Create(ctx, orgRef, repoInfo, repoCreateOpts)
+		if err != nil && strings.Contains(err.Error(), "rate limit exceeded") {
+			waitForRateQuota, err := getWaitTimeFromErr(err.Error())
+			if err != nil {
+				return err
+			}
+			fmt.Printf("Waiting for rate quota %s \n", waitForRateQuota.String())
+			time.Sleep(waitForRateQuota)
+			return fmt.Errorf("retry after waiting for rate quota")
+		}
+		return err
+	}); err != nil {
+		return fmt.Errorf("error creating repo %s", err)
+	}
+	fmt.Printf("repo %s created ...\n", repoName)
+
+	fmt.Printf("validating access to the repo %s ...\n", repoName)
+	err = utils.WaitUntil(os.Stdout, time.Second, time.Second*30, func() error {
```

Comment:
If you turn time.Second*30 into a const per the above suggestion then you can update it here too.